source/module_hsolver/kernels/cuda/math_kernel_op.cu

```diff
@@ -93,13 +93,24 @@ __global__ void line_minimize_with_block(
     }
     __syncthreads();
     // just do some parallel reduction in shared memory
-    for (int ii = THREAD_PER_BLOCK >> 1; ii > 0; ii >>= 1) {
+    for (int ii = THREAD_PER_BLOCK >> 1; ii > WARP_SIZE; ii >>= 1) {
```

Comment:
In this PR, you have optimized the performance of this code, please add annotations to explain the reason.